From 1d9b60f11a2a1e7a6784fcca8f3f5c0924343314 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:00:19 -0400
Subject: [ethernet] mlx4: Call dev_kfree_skb_any instead of dev_kfree_skb

Message-id: <65fee1c30432513638771ce90e827bfe14d94b13.1410425016.git.avadai@redhat.com>
Patchwork-id: 93618
O-Subject: [Patch RHEL7.1 V1 083/197] mlx4: Call dev_kfree_skby_any instead of dev_kfree_skb.
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: "Eric W. Biederman" <ebiederm@xmission.com>

Replace dev_kfree_skb with dev_kfree_skb_any in functions that can
be called in hard irq and other contexts.

Signed-off-by: "Eric W. Biederman" <ebiederm@xmission.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit e81f44b66b456a7dcfbdeffeb355458cd6a58973)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_tx.c b/drivers/net/ethernet/mellanox/mlx4/en_tx.c
index 9cdeca4..7ee18be 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_tx.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_tx.c
@@ -314,7 +314,7 @@ static u32 mlx4_en_free_tx_desc(struct mlx4_en_priv *priv,
    }
   }
  }
- dev_kfree_skb(skb);
+ dev_kfree_skb_any(skb);
  return tx_info->nr_txbb;
 }
 
-- 
1.7.1